core/trino-main/src/main/java/io/trino/connector/system/MaterializedViewSystemTable.java

```diff
@@ -0,0 +1,145 @@
+            .column("comment", createUnboundedVarcharType())
+            .column("definition", createUnboundedVarcharType())
+            .build();
+
+    private final Metadata metadata;
+    private final AccessControl accessControl;
+
+    @Inject
+    public MaterializedViewSystemTable(Metadata metadata, AccessControl accessControl)
+    {
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return TABLE_DEFINITION;
+    }
+
+    @Override
+    public RecordCursor cursor(
+            ConnectorTransactionHandle transactionHandle,
+            ConnectorSession connectorSession,
+            TupleDomain<Integer> constraint)
+    {
+        Session session = ((FullConnectorSession) connectorSession).getSession();
+        InMemoryRecordSet.Builder displayTable = InMemoryRecordSet.builder(getTableMetadata());
+        getAllMaterializedViews(session, constraint, displayTable);
+        return displayTable.build().cursor();
+    }
+
+    private void getAllMaterializedViews(Session session, TupleDomain<Integer> constraint, InMemoryRecordSet.Builder displayTable)
+    {
+        Optional<String> catalogNameFilter = tryGetSingleVarcharValue(constraint, 0);
+        Optional<String> schemaNameFilter = tryGetSingleVarcharValue(constraint, 1);
+        Optional<String> tableNameFilter = tryGetSingleVarcharValue(constraint, 2);
+
+        listCatalogs(session, metadata, accessControl, catalogNameFilter).keySet().forEach(catalogName -> {
+            QualifiedTablePrefix tablePrefix = tablePrefix(catalogName, schemaNameFilter, tableNameFilter);
+
+            getMaterializedViews(session, metadata, accessControl, tablePrefix).forEach((tableName, definition) -> {
+                MaterializedViewInfo materializedView = createMaterializedView(session, new CatalogSchemaTableName(tablePrefix.getCatalogName(), tableName), definition);
+                displayTable.addRow(toTableRow(materializedView));
+            });
+        });
+    }
+
+    protected MaterializedViewInfo createMaterializedView(
+            Session session,
+            CatalogSchemaTableName tableName,
+            ConnectorMaterializedViewDefinition definition)
+    {
+        QualifiedObjectName name = new QualifiedObjectName(
+                tableName.getCatalogName(),
+                tableName.getSchemaTableName().getSchemaName(),
+                tableName.getSchemaTableName().getTableName());
+
+        MaterializedViewFreshness freshness = metadata.getMaterializedViewFreshness(session, name);
+
+        return new MaterializedViewInfo(name, definition, freshness);
+    }
+
+    protected Object[] toTableRow(MaterializedViewInfo materializedView)
```

Comment:
We can now convert to Object[] from tableName, definition and freshness directly, I don't see a need for MaterializedViewInfo class now.